src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/DoNotPrefixEnumValuesWithTypeName.cs

```diff
@@ -0,0 +1,66 @@
+    /// CA1712: Do not prefix enum values with type name
+    /// </summary>
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public sealed class DoNotPrefixEnumValuesWithTypeNameAnalyzer : DiagnosticAnalyzer
+    {
+        internal const string RuleId = "CA1712";
+        private const int PercentValuesPrefixedThreshold = 75; // The percent of an enum's values that must appear to be prefixed in order for a diagnostic to be reported on the enum. This value comes from the original FxCop rule's implementation.
+
+        private static readonly LocalizableString s_localizableTitle = new LocalizableResourceString(nameof(DoNotPrefixEnumValuesWithTypeNameTitle), ResourceManager, typeof(MicrosoftApiDesignGuidelinesAnalyzersResources));
+        private static readonly LocalizableString s_localizableMessage = new LocalizableResourceString(nameof(DoNotPrefixEnumValuesWithTypeNameMessage), ResourceManager, typeof(MicrosoftApiDesignGuidelinesAnalyzersResources));
+        private static readonly LocalizableString s_localizableDescription = new LocalizableResourceString(nameof(DoNotPrefixEnumValuesWithTypeNameDescription), ResourceManager, typeof(MicrosoftApiDesignGuidelinesAnalyzersResources));
+
+        internal static DiagnosticDescriptor Rule =
+            new DiagnosticDescriptor(
+                RuleId,
+                s_localizableTitle,
+                s_localizableMessage,
+                DiagnosticCategory.Naming,
+                DiagnosticSeverity.Error,
+                isEnabledByDefault: true,
+                description: s_localizableDescription,
+                helpLinkUri: "https://msdn.microsoft.com/en-us/library/ms182237.aspx",
+                customTags: FxCopWellKnownDiagnosticTags.PortedFxCopRule);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext analysisContext)
+        {
+            analysisContext.EnableConcurrentExecution();
+            analysisContext.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            analysisContext.RegisterSymbolAction(AnalyzeNamedType, SymbolKind.NamedType);
+        }
+
+        private static void AnalyzeNamedType(SymbolAnalysisContext context)
+        {
+            var symbol = (INamedTypeSymbol)context.Symbol;
+            IEnumerable<ISymbol> enumValues;
+            if (symbol.TypeKind != TypeKind.Enum || !(enumValues = symbol.GetMembers().Where(m => m.Kind == SymbolKind.Field)).Any())
+            {
+                return;
+            }
+
+            int numPrefixed = enumValues.Count(m => m.Name.StartsWith(symbol.Name, StringComparison.OrdinalIgnoreCase));
+            int percentPrefixed = 100 * numPrefixed / enumValues.Count();
+            if (percentPrefixed >= PercentValuesPrefixedThreshold)
+            {
+                context.ReportDiagnostic(symbol.CreateDiagnostic(Rule, symbol.Name));
```

Comment:
We should still report diagnostic for each field, not on the enum.

Reply:
It seems unnecessarily noisy to report it on every field name, especially when the heuristic aims to only let this fire when a majority of the fields need fixing. I.e. you'll either have no diagnostics, or multiple, nearly identical diagnostics.  However, I'll make the change, based on the compatibility argument.